specs/permissions-deployments.md

```diff
@@ -0,0 +1,72 @@
+
+The provisioning info file is a single file for all permissions. Permissions are still the primary concept with a many to many relationship between permissions and cloud environments. Permission claim values are the top level members to enable quick lookup for provisioning for a given permission. Not all implemented permission schemes are published across all environments. The same permission would have different GUIDs in different environments and even when all schemes of a permission are published to the same environment, they would happen to have different unique identifiers. The provisioning info file allows us to model the cloud availability of a permission across different clouds for each of its schemes.
+
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14 [RFC2119] [RFC8174] when, and only when, they appear in all capitals, as shown here.
+
+## <a name="permissionsDeploymentDocumentObject"></a> PermissionsDeployments Document Object
+
+This is the top level object in the provisioning info file. It contains a "permissionDeployments" property that contains a JSON object.
+
+```json
+{
+    "permissionDeployments": {
+        "Files.Read.All": [
+            {
+                "id": "bed71653-0fa8-42a0-b33c-a1aab02ecda6",
+                "environment": "global",
+                "scheme": "DelegatedWork",
+                "isHidden": false,
+                "isDisabled": false,
+                "resourceAppId": "00000003-0000-0000-c000-000000000000"
+            },
+            {
+                "id": "bed71753-0fq8-42a2-b33c-c1acb02ecd17",
+                "environment": "global",
+                "scheme": "Application",
+                "isHidden": false,
+                "isDisabled": false
+            }
+        ]
+    }
+}
+```
+
+In this example, the permission "Files.Read.All" is present in the "global" environment for both "Delegated" and "Application" schemes. 
+
+### permissionDeployments Object
+
+The "permissionDeployments" object contains a map of  permission names to "permissionDeployment" objects.
+
+### <a name="permissionDeploymentObject"></a>PermissionDeployment Object
+
+This is a JSON object that describes where a permission has been deployed and its current state.
+
+#### id
+The "id" member is the GUID of the permission deployment.
+
+#### environment member
+
+The "environment" member is a JSON string that identifies the deployment environment in which the permission SHOULD be supported.
+
+#### scheme member
+The "scheme" member is a JSON string that idenfies the type of authorization scheme that is enabled for the containing permission.
+
+#### isHidden
+The "isHidden" member is a boolean value that indicates if a permission should be publicly usable in the API.
+
+#### isDisabled
```

Comment:
Can we change this to isEnabled? It's the same as what we have in the SP response https://learn.microsoft.com/en-us/graph/api/serviceprincipal-get?view=graph-rest-1.0&tabs=http#response-2

Reply:
Let me update this in https://github.com/microsoftgraph/kibali/pull/42

Reply:
Having isEnabled means that the property must be present in every deployment object.  If it was isDisabled then we only need to include where they are actually disabled, which should be just a small number.